omnipaxos/src/sequence_paxos/follower.rs

```diff
@@ -184,17 +132,22 @@ where
             && self.state.1 == Phase::Accept
             && self.handle_sequence_num(dec.seq_num, dec.n.pid) == MessageStatus::Expected
         {
-            if dec.decided_idx > self.internal_storage.get_accepted_idx() {
-                let new_accepted_idx = self.internal_storage.flush_batch().expect(WRITE_ERROR_MSG);
-                self.handle_flushed_accepted(dec.n, new_accepted_idx);
+            let mut new_decided_idx = dec.decided_idx;
```

Comment:
Becomes this:
```rust
    pub(crate) fn handle_acceptdecide(&mut self, acc_dec: AcceptDecide<T>) {
        if self.check_valid_ballot(acc_dec.n)
            && self.state == (Role::Follower, Phase::Accept)
            && self.handle_sequence_num(acc_dec.seq_num, acc_dec.n.pid) == MessageStatus::Expected
        {
            #[cfg(not(feature = "unicache"))]
            let entries = acc_dec.entries;
            #[cfg(feature = "unicache")]
            let entries = self.internal_storage.decode_entries(acc_dec.entries);
            let mut new_accepted_idx = self
                .internal_storage
                .append_entries_and_get_accepted_idx(entries)
                .expect(WRITE_ERROR_MSG);
            new_accepted_idx = new_accepted_idx.max(self.update_decided_idx_and_get_accepted_idx(acc_dec.decided_idx));
            if let Some(idx) = new_accepted_idx {
                self.reply_accepted(acc_dec.n, idx);
            }
        }
    }
```
Kinda strange imo, but maybe you like it?

Reply:
Correct me if I'm wrong, but I don't think the first `new_accepted_idx` can be greater than the second (the one after maybe flushing when deciding)? I think if we remove the `max()` it will be less confusing.

```rust
let mut new_accepted_idx: Option<usize> = self.internal_storage.append_entries_and_get_accepted_idx(...).expect(...);
let flushed_after_decide: Option<usize> = self.update_decided_idx_and_get_accepted_idx(...);
if flushed_after_decide.is_some() {
    new_accepted_idx = flushed_after_decide;
}
if let Some(idx) = new_accepted_idx {
    self.reply_accepted(acc_dec.n, idx);
}
 

```

Reply:
It's there because the calculated new_accepted_idx are Options (since we don't want to send Accepted if nothing was flushed). So:
None.max(Some(flush_idx)) = Some(flush_idx) for example.